overlord/devicestate/devicestate_install_mode_test.go

```diff
@@ -486,6 +493,53 @@ func (s *deviceMgrInstallModeSuite) TestInstallWithInstallDeviceHookExpTasks(c *
 	c.Assert(hooksCalled[0].HookName(), Equals, "install-device")
 }
 
+func (s *deviceMgrInstallModeSuite) testInstallWithInstallDeviceHookSnapctlReboot(c *C, arg string, rst state.RestartType) {
+	restore := release.MockOnClassic(false)
+	defer restore()
+
+	restore = devicestate.MockInstallRun(func(mod gadget.Model, gadgetRoot, kernelRoot, device string, options install.Options, _ gadget.ContentObserver) (*install.InstalledSystemSideData, error) {
+		return nil, nil
+	})
+	defer restore()
+
+	restore = hookstate.MockRunHook(func(ctx *hookstate.Context, tomb *tomb.Tomb) ([]byte, error) {
+		c.Assert(ctx.HookName(), Equals, "install-device")
+
+		// snapctl reboot --halt
+		_, _, err := ctlcmd.Run(ctx, []string{"reboot", arg}, 0)
```

Comment:
to be clear, in this suite we have already mocked the reboot machinery such that this won't end up calling the real `reboot` command in response to this line of code calling `snapctl reboot <arg>` correct ?

Reply:
this simulates "snapctl reboot", the real reboot happens via daemon, which here is not wired up at all. What gets wired up is state.RequestRestart that,  you can see that in devicestate_test.go:

`s.o = overlord.MockWithStateAndRestartHandler(nil, func(req state.RestartType) ...`

and feeds into the suite restartRequests []state.RestartType  field

Reply:
ah right I forgot that it doesn't go that far to daemon, thanks for explaining